diff --git a/Package.swift b/Package.swift
index 38e51f6317..8f15002268 100644
--- a/Package.swift
+++ b/Package.swift
@@ -28,7 +28,7 @@ let package = Package(
     .package(path: "../swift-custom-dump"),
     .package(path: "../swift-dependencies"),
     .package(path: "../swift-identified-collections"),
-    .package(url: "https://github.com/pointfreeco/swift-macro-testing", from: "0.2.0"),
+
     .package(path: "../swift-navigation"),
     .package(path: "../swift-perception"),
     .package(path: "../swift-sharing"),
@@ -88,13 +88,6 @@ let package = Package(
         .product(name: "SwiftCompilerPlugin", package: "swift-syntax"),
       ]
     ),
-    .testTarget(
-      name: "ComposableArchitectureMacrosTests",
-      dependencies: [
-        "ComposableArchitectureMacros",
-        .product(name: "MacroTesting", package: "swift-macro-testing"),
-      ]
-    ),
   ]
 )
 
diff --git a/Package@swift-6.0.swift b/Package@swift-6.0.swift
index b8d82fc893..dd4f687a15 100644
--- a/Package@swift-6.0.swift
+++ b/Package@swift-6.0.swift
@@ -24,11 +24,11 @@ let package = Package(
     .package(url: "https://github.com/OpenCombine/OpenCombine.git", from: "0.14.0"),
     .package(path: "../combine-schedulers"),
     .package(path: "../swift-case-paths"),
-    .package(url: "https://github.com/pointfreeco/swift-concurrency-extras", from: "1.2.0"),
+    .package(path: "../swift-concurrency-extras"),
     .package(path: "../swift-custom-dump"),
     .package(path: "../swift-dependencies"),
     .package(path: "../swift-identified-collections"),
-    .package(url: "https://github.com/pointfreeco/swift-macro-testing", from: "0.2.0"),
+    .package(path: "../swift-macro-testing"),
     .package(path: "../swift-navigation"),
     .package(path: "../swift-perception"),
     .package(path: "../swift-sharing"),
